eclipse-che/che-server
# fix: Adding ability to use kubernetes API proxy on Kubernetes

Signed-off-by: xbaran4 <pbaran@redhat.com>



### What does this PR do?
Adds property, that will allow the use of /unsupported/k8s even when on Kubernetes infrastructure.

### Screenshot/screencast of this PR


### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19813

### How to test this PR?

* run tests
1. have or deploy che instance on minikube with chectl
2. run
```shell
kubectl patch checluster/eclipse-che -n eclipse-che --type=merge -p "$(cat patch.yaml)"
```
where patch.yaml contains:
```
spec:
  server:
    cheImage: "quay.io/pbaran/che-server"
    cheImageTag: "unsupported-k8s"
```
3. Enable unsupported k8s api on k8s
```
kubectl patch checluster/eclipse-che --patch "{\"spec\":{\"server\":{\"customCheProperties\": {\"CHE_INFRA_KUBERNETES_ENABLE__UNSUPPORTED__K8S\": \"true\"}}}}" --type=merge -n eclipse-che
```
3.run
```shell
curl -k ${CHE_HOST}/api/unsupported/k8s/api/v1/namespaces?token=${TOKEN}
```
where CHE_HOST is the URL of your che-server instance and TOKEN is your users keycloak token

4. you should see something similar to this: 
![image](https://user-images.githubusercontent.com/73115616/119663589-c48ab900-be32-11eb-9af6-e9bc0cf57226.png)

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
